infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/PodMergerTest.java

```diff
@@ -237,6 +237,57 @@ public void shouldNotAddImagePullPolicyTwice() throws Exception {
     assertEquals(imagePullSecrets.get(0).getName(), "secret");
   }
 
+  @Test
+  public void shouldMergeTerminationGracePeriodSharedByPods() throws Exception {
+    // given
+    PodSpec podSpec1 = new PodSpecBuilder().withTerminationGracePeriodSeconds(24L).build();
+    PodData podData1 = new PodData(podSpec1, new ObjectMetaBuilder().build());
+    PodSpec podSpec2 = new PodSpecBuilder().withTerminationGracePeriodSeconds(24L).build();
+    PodData podData2 = new PodData(podSpec2, new ObjectMetaBuilder().build());
+
+    // when
+    Deployment merged = podMerger.merge(Arrays.asList(podData1, podData2));
+
+    // then
+    PodTemplateSpec podTemplate = merged.getSpec().getTemplate();
+    assertEquals(podTemplate.getSpec().getTerminationGracePeriodSeconds(), (Long) 24L);
+  }
+
+  @Test
+  public void shouldMergeTerminationGracePeriodSharedByPodsIfOneIsNull() throws Exception {
+    // given
+    PodSpec podSpec1 = new PodSpecBuilder().withTerminationGracePeriodSeconds(null).build();
+    PodData podData1 = new PodData(podSpec1, new ObjectMetaBuilder().build());
+    PodSpec podSpec2 = new PodSpecBuilder().withTerminationGracePeriodSeconds(24L).build();
+    PodData podData2 = new PodData(podSpec2, new ObjectMetaBuilder().build());
+
+    // when
+    Deployment merged = podMerger.merge(Arrays.asList(podData1, podData2));
+
+    // then
+    PodTemplateSpec podTemplate = merged.getSpec().getTemplate();
+    assertEquals(podTemplate.getSpec().getTerminationGracePeriodSeconds(), (Long) 24L);
+  }
+
+  @Test(
+      expectedExceptions = ValidationException.class,
+      expectedExceptionsMessageRegExp =
+          "Cannot merge pods with a different configuration of the termination grace period: 24, 25")
+  public void shouldFailIfTerminationGracePeriodSharedBDiffersInPods() throws Exception {
+    // given
+    PodSpec podSpec1 = new PodSpecBuilder().withTerminationGracePeriodSeconds(24L).build();
+    PodData podData1 = new PodData(podSpec1, new ObjectMetaBuilder().build());
+    PodSpec podSpec2 = new PodSpecBuilder().withTerminationGracePeriodSeconds(25L).build();
+    PodData podData2 = new PodData(podSpec2, new ObjectMetaBuilder().build());
+
+    // when
+    Deployment merged = podMerger.merge(Arrays.asList(podData1, podData2));
+
+    // then
+    PodTemplateSpec podTemplate = merged.getSpec().getTemplate();
+    assertEquals(podTemplate.getSpec().getTerminationGracePeriodSeconds(), (Long) 42L);
```

Comment:
why this assert? shouldn't it throw an exception during the merge on Line 284 and end there?